### ISSUE TITLE
Sometimes, movement will be forced in one direction until you reconnect or spam as hard as you can to get out of it

### ISSUE THREAD

**Issue by Mothblocks** [tgstation/tgstation#52706]
Sometimes, when you move, you'll be locked in that direction. You then either have to reconnect, or get lucky with spamming the movement keys.

This is happening significantly more frequently on Bagil for me and purportedly others, and I've personally had it happen to me 5 times per round.

It's not related to tgchat, as I have received information that it has been reproduced on Sybil.

**Comment by optimumtact** [comment]
This is due to the key getting stuck in the users client keys_held, even though they are not holding down the button.

You can clear it as an admin by VV the user and clearing the keys_held var

I guesttimate it's due to key_up not being called correctly, but I've never been able to reliably replicate in order to identify the root cause.

**Comment by tralezab** [comment]
This is from opening a box in the tgui window while holding down a key, as far as i'm aware

**Comment by Tlaltecuhtli** [comment]
i think it happens more commonly when u use windows with choices or text field (not tgui)

**Comment by necromanceranne** [comment]
Yeah it's textfield stuff, I've had this happen when using those. I've managed to fix it a few times but I have no idea how I managed it. Only by reconnecting have I ever got it to fix.

**Comment by san7890** [comment]
Sometimes, if you remember what you did to cause yourself to get stuck your movement, and you repeat the same exact action, it'll unstick itself. I got this tactic to work about 80% of the time during normal gameplay, but a reconnect/reboot is pretty much the guaranteed way to get out of walking and walking and walking and walking.......

**Comment by Tharcoonvagh** [comment]
Really annoying when playing as the ai

**Comment by optimumtact** [comment]
I was able to trigger this as a ghost by clicking on a camera console while moving does the AI have an action button for a camera console?

probably caused by a similar hijacking of input meaning the keydown never arrives.

**Comment by ZeroMODE0** [comment]
Sometimes this happens to me as Medborg or AI when using letters to sift through cameras/synthesizer menus. Example: A for Aiuri pressed, hit enter, forever move left until A is mashed or you get spaced because you happened to be in line with an airlock. Holding D will keep me still, and S and W still move me up and down.

**Comment by Couls** [comment]
#53773 will help with this but unsure if it will entirely fix it

**Comment by Dennok** [comment]
Just clear  push/unpush button don't help? It help me sometimes but you need click on game window (spacemen, turfs, etc)
